File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.js
// Last updated: 4/29/2026, 10:25:14 AM
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function(n) {
    const map = {};
    const dp = i => {
        if(map[i] !== undefined) {
            return map[i];
        }
        if(i === 0) {
            return 0;
        } else if(i <= 2) {
            return 1;
        }
        map[i] = dp(i-1) + dp(i-2) + dp(i-3);
        return map[i];
    }
    return dp(n);
};

File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.py
# Last updated: 4/29/2026, 10:25:19 AM
class Solution:
    def tribonacci(self, n: int) -> int:
        @cache
        def tribonacci(i):
            if i <= 1:
                return i
            elif i == 2:
                return 1
            return tribonacci(i-3) + tribonacci(i-2) + tribonacci(i-1)
        
        return tribonacci(n)